lib/home.dart

```diff
@@ -227,6 +297,7 @@ class _HomeState extends State<Home> {
                 },
                 tabBar: CupertinoTabBar(
                     items: _navigationItems,
+                    currentIndex: _controller.index,
                     onTap: (index) {
```

Comment:
If I understand correctly, it seems we only need to add a **save to local** operation here?

So `didChangeAppLifecycleState` and `didChangeDependencies` stuffs could be removed.

Reply:
So you're telling I should perform writing to the storage during `onTap` itself? I thought that it would mean writing it too many times. Also, `didChangeDependencies` is for loading the values initially. So that has to be there.

Reply:
> Also, didChangeDependencies is for loading the values initially. So that has to be there.

Perhaps we could move the initial read to `themeModel.init()`? This method would be called before rendering.

https://github.com/git-touch/git-touch/blob/be30bcb7d8ea29ca96c30b5d24e2ce87d8caad2e/lib/main.dart#L39

Reply:
But how will I set `active` and `_controller.index` in `themeModel.init`?

Reply:
I think there are two ways to do it:

1. Get this value from `app.dart` and pass it to `home.dart` as props, then assign it to active as initial value: `int active = widget.initialActive`
2. Move the `active` state to theme model

Reply:
Alright thanks a lot. Second one seems easier. I'll try it out.

> So you're telling I should perform writing to the storage during `onTap` itself?

Also this?

Reply:
Oh, I just learned what `didChangeAppLifecycleState` lifecycle does.

I think both would be OK.

Reply:
For me it seems simple to understand to save it on tap callback.

If the way you mentioned could make it more performant, I think it is also acceptable.

Reply:
I put in onTap itself. Might make a minor improvement in performance, but your way makes it easier to understand what's happening.